PyTorch Autolog Example Readme files

karthik-77 opened the pull request:
…cation and Mnist example file

Signed-off-by: karthik-77 <karthiks@ideas2it.com>

## What changes are proposed in this pull request?

Updated readme files for the pytorch autolog examples :bert-news-classification,mnist -example1,example2

## How is this patch tested?

Documentation - NA

## Release Notes

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

karthiks-77 committed:
Added the pytorch autolog example readme files for bert-news-classification and Mnist example file

Signed-off-by: karthik-77 <karthiks@ideas2it.com>

smurching committed:
Update README.md

smurching committed:
Update README.md

smurching commented on examples/pytorch/BertNewsClassification/README.md:
```diff
@@ -1,22 +1,5 @@
-Example: `export MLFLOW_TRACKING_URI=http://localhost:5000/`
-
-For more details - https://mlflow.org/docs/latest/tracking.html#where-runs-are-recorded
+## BERT news classification example
```
@karthik-77 I reworked this file with a couple of changes:
* Simplified the intro & removed the "code related to MLflow" section, which is redundant given the description of ``mlflow.pytorch.autolog`` here
* Split out the ``mlflow ui`` command into its own "Viewing results in the UI" section and moved it above the description of how to customize training parameters (to streamline the common path of running the example with default parameters -> viewing results in the UI).
* Reworded the description of training parameters 

Would you be able to apply similar changes to the other READMEs? Thanks!

karthik-77 replied:
I have applied similar changes in the other read me files (Mnist-example1,example2) and updated the PR

karthik-77 commented:
Hi Siddarth,

Sure i will rework on the same lines, and make the changes accordingly in
the respective files.

Thanks and Regards
karthik

On Thu, Nov 5, 2020 at 10:14 AM Siddharth Murching <notifications@github.com>
wrote:

> *@smurching* commented on this pull request.
> ------------------------------
>
> In examples/pytorch/BertNewsClassification/README.md
> <https://github.com/mlflow/mlflow/pull/3639#discussion_r517789552>:
>
> > -The code is trained using pytorch lightning loop and the autolog function call in the main - `mlflow.pytorch.autolog()`
> -is responsible for logging the params, metrics, model summary and the model.
> -
> -
> -### Code related to MLflow:
> -* [`mlflow.pytorch.autolog`]
> -This is an experimental api that logs ML model artifacts and metrics.
> -The metrics are logged during training of the model.
> -
> -## Setting tracking URI
> -MLflow tracking URI can be set using the environment variable MLFLOW_TRACKING_URI
> -
> -Example: `export MLFLOW_TRACKING_URI=http://localhost:5000/` <http://localhost:5000/>
> -
> -For more details - https://mlflow.org/docs/latest/tracking.html#where-runs-are-recorded
> +## BERT news classification example
>
> @karthik-77 <https://github.com/karthik-77> I reworked this file with a
> couple of changes:
>
>    - Simplified the intro & removed the "code related to MLflow" section,
>    which is redundant given the description of mlflow.pytorch.autolog here
>    - Split out the mlflow ui command into its own "Viewing results in the
>    UI" section and moved it above the description of how to customize training
>    parameters (to streamline the common path of running the example with
>    default parameters -> viewing results in the UI).
>    - Reworded the description of training parameters
>
> Would you be able to apply similar changes to the other READMEs? Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mlflow/mlflow/pull/3639#pullrequestreview-523921948>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARRYHDCJEWAMUGWYQA7HP5DSOIUTXANCNFSM4TKF3N5Q>
> .
>

smurching committed:
Update README.md

Add example of passing custom parameters when running example as an MLflow project

smurching commented on examples/pytorch/BertNewsClassification/README.md:
````diff
@@ -41,35 +24,45 @@ If you have the required modules for the file and would like to skip the creatio
 
 For example:
 ```
+mlflow run . -P epochs=5 -P gpus=1 -P batch_size=32 -P num_workers=2 -P learning_rate=0.01 -P accelerator="ddp"
````
@karthik-77 I added an example here of how to run the training script as an MLflow project with custom parameters, for consistency / logical flow from the previous section, which explains how to run training script as a project. Could we add a similar section to the other READMEs as well?

smurching reviewed (commented):
Thanks @karthik-77 for the PR! I noticed a few other issues with the BERT example that it'd be great to address in another PR:

* We have a parameter called ``max_epochs`` in the script, but ``epochs`` in the MLproject file. Could we make the two consistent (e.g. ``max_epochs`` in both places?)
* There’s a ``num_workers`` param documented to have a default of 0 [here](https://github.com/mlflow/mlflow/blob/055978cdd425ac7c328069d80230f5bc50265df8/examples/pytorch/BertNewsClassification/bert_classification.py#L172), but the default is really 3. Could we update the linked line with the correct default value?

karthik-77 commented:
Hi Siddarth,

Sure will make those changes so that consistency is maintained throughout.
Thanks for pointing that out.

Thanks and Regards
karthik

On Thu, Nov 5, 2020 at 10:21 AM Siddharth Murching <notifications@github.com>
wrote:

> *@smurching* commented on this pull request.
>
> Thanks @karthik-77 <https://github.com/karthik-77> for the PR! I noticed
> a few other issues with the BERT example that it'd be great to address in
> another PR:
>
>    - We have a parameter called max_epochs in the script, but epochs in
>    the MLproject file. Could we make the two consistent (e.g. max_epochs
>    in both places?)
>    - There’s a num_workers param documented to have a default of 0 here
>    <https://github.com/mlflow/mlflow/blob/055978cdd425ac7c328069d80230f5bc50265df8/examples/pytorch/BertNewsClassification/bert_classification.py#L172>,
>    but the default is really 3. Could we update the linked line with the
>    correct default value?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mlflow/mlflow/pull/3639#pullrequestreview-523923837>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARRYHDHSZXAKYTWNB5JDAQLSOIVMFANCNFSM4TKF3N5Q>
> .
>

karthik-77 commented:
@smurching  Request you to kindly sign off the commits.

karthiks-77 committed:
BERT news classification and MNIST example 1 and 2 folders have been updated according to comments

Signed-off-by: karthik-77 <karthiks@ideas2it.com>

smurching committed:
Fix typos & spelling mistakes

Signed-off-by: Sid Murching <sid.murching@databricks.com>

smurching reviewed (approved):
LGTM
